qemu/tests/sgx_cpu.py

```diff
@@ -0,0 +1,51 @@
+from virttest import error_context
+from virttest.utils_misc import verify_dmesg
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Qemu sgx cpu test:
+    1. Boot sgx VM
+    2. Check sgx guest cpuid with sub-features
+    3. Check sgx guest corresponding CPUID entries
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    error_context.context("Start sgx cpu test", test.log.info)
+    timeout = params.get_numeric("login_timeout", 240)
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login(timeout=timeout)
+    verify_dmesg()
+
+    # Set up guest environment
+    cpuid_pkg = params.get("cpuid_pkg")
+    if session.cmd_status("rpm -qa|grep %s" % cpuid_pkg):
+        try:
+            session.cmd_output_safe(params.get("repo_install_cmd"))
+            session.cmd_status("yum -y install %s" % cpuid_pkg)
+        except Exception:
+            test.cancel("Fail to install package cpuid, please retest this "
```

Comment:
I think we should add session.close() before test.cancel. Thanks.

Reply:
Thanks Xueqiang for pointing it out, yes, it's better to release all resources after finishing our testing. Keep this in mind, then we can avoid resource leaks (esp. the images).
But for session, I don't stick at that because finally all sessions can be released after vm shutdown, so it's not harmful to our testing env.
In a summary, it's better to release all resources(recommended!), but actually, for some resources, eg. session, it's OK to release it after vm shutdown  cc @XueqiangWei  @aliang123

Reply:
@zhencliu Thanks Zhenchao. Get it, I just want to add a reminder. I once hit memory leak because the vm was not destroyed successfully, and a large number of sessions were not closed.

Reply:
Thanks @XueqiangWei for the reminder!

Reply:
Thanks @XueqiangWei and @zhencliu I add session.close() before test.cancel

Reply:
hi @zixi-chen , if you'd like to add session.close, I am afraid you have to consider all return points, like line42 and line50, so you need a try...finally... outside all code, and close the session and destroy the vm in the finally clause.

Reply:
@zhencliu do we need to try finally all code, I think we only need to add try: after session = vm.wait_for_login(timeout=timeout), and vm.destroy () still in the last line but outside finally:

Reply:
Sure, you are right. Actually vm can be destroyed by VT, but it's OK for me to explicitly destroy the VM after our test @zixi-chen

Reply:
Thanks, Update the patch